### PR TITLE
docs: add contributor notes on where to find documentation branches

### PR DESCRIPTION
**Pull request by carlsonp**

### CODE REVIEW

**Review by taylorfturner — CHANGES_REQUESTED** [review]
open into `dev` branch instead of `main` but otherwise LGTM

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -106,3 +106,6 @@ the documentation.
 
 ## Updating Dependencies
 If you make changes to the `requirements` text files, please also update the `additional_dependencies` list under the `mypy` hook in `.pre-commit-config.yaml`. This is necessary for accurate type-checking.
+
+## Contributing Documentation Changes and Fixes
+Please see the `gh-pages` and `dev-gh-pages` branches for all documentation.
```
Comment (taylorfturner) [inline .github/CONTRIBUTING.md:111]: Technically a PR for updating the documentation should only go into `dev-gh-pages` -- and when we do a release a PR will be opened to `gh-pages`.

**Review by taylorfturner — APPROVED** [review]
LGTM -- small comment for follow-up PR. Thanks for the contribution @carlsonp

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -106,3 +106,8 @@ the documentation.
 
 ## Updating Dependencies
 If you make changes to the `requirements` text files, please also update the `additional_dependencies` list under the `mypy` hook in `.pre-commit-config.yaml`. This is necessary for accurate type-checking.
+
+## Contributing Documentation Changes and Fixes
+When making adjustments or contributions to documentation, please use the `dev-gh-pages` branch.  This is where all the documentation lives.
+After you've completed your edits, open a Github Pull Request (PR) to merge into `dev-gh-pages`.  During a version release, `dev-gh-pages` is merged
```
Comment (taylorfturner) [inline .github/CONTRIBUTING.md:112]:
```suggestion
After you've completed your edits, open a Github Pull Request (PR) to merge into `dev-gh-pages` from your fork.  During a version release, `dev-gh-pages` is merged
```

**Review by taylorfturner — CHANGES_REQUESTED** [review]
slightly nuance clarification

**Comment by taylorfturner** [comment]
Seeing a test error: `E   ValueError: Must install dask-expr to activate query planning` probably related to `E   ModuleNotFoundError: No module named 'dask_expr'`. Not seeing this on other PRs at the moment, but dask did make a release yesterday https://pypi.org/project/dask/#history

**Comment by taylorfturner** [comment]
I approve this change, @carlsonp  -- we'll just need this dask issue resolved so tests pass on the checks. If you have time, welcome to research and propose a fix. Otherwise, I'll have to see if I can get to it later; but this change itself is good! Thanks for the contributions @carlsonp!

**Comment by taylorfturner** [comment]
@carlsonp you should be good to rebase onto `dev` now

**Comment by carlsonp** [comment]
Rebased

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -106,3 +106,8 @@ the documentation.
 
 ## Updating Dependencies
 If you make changes to the `requirements` text files, please also update the `additional_dependencies` list under the `mypy` hook in `.pre-commit-config.yaml`. This is necessary for accurate type-checking.
+
+## Contributing Documentation Changes and Fixes
+When making adjustments or contributions to documentation, please use the `dev-gh-pages` branch.  This is where all the documentation lives.
+After you've completed your edits, open a Github Pull Request (PR) to merge into `dev-gh-pages` from your fork.  During a version release, `dev-gh-pages` is merged
+into the `gh-pages` branch and is the version associated with the documentation website and stable version.
```
Comment (taylorfturner) [inline .github/CONTRIBUTING.md:113]:
```suggestion
into the `gh-pages` branch (after `update_documentation.py` is run) and is the version associated with the documentation website and stable version.
```

**Review by taylorfturner — COMMENTED** [review]
slight update @carlsonp

**Review by taylorfturner — APPROVED** [review]
LGTM

**Review by ksneab7 — APPROVED** [review]

**Comment by taylorfturner** [comment]
@carlsonp needs update branch now after #1114  merged

**Pull request merged**